### ISSUE TITLE
Cross Entropy tests are wrong

### ISSUE THREAD

**Issue by 0xgeorgeassaad** [Machine-Learning-Foundations/exercise_10_neural_networks#9]
First of all, Thanks for making this course open-source.

Correct me If I am wrong but I think the test cases for the cross entropy loss are wrong.

Here are both test cases and their outputs using Jax and PyTorch and my implementation. 
The outputs of my implementation, Jax, and Pytorch functions are the same but they don't match the expected result.

Also, there is an issue with the labels themselves as each row should be a valid probability distribution (non-negative, sums to 1) but there are some rows that sums to 0.0

```python
import jax.numpy as jnp
import optax

def cross_entropy(labels: jnp.ndarray, logits: jnp.ndarray) -> jnp.ndarray:
    """Compute the cross entropy of one-hot encoded labels and the logits."""
    out = nn.log_softmax(logits, axis=1)
    return -jnp.mean(jnp.sum(label * out, axis=-1))



# test case 1
labels = jnp.array([[1.0, 0.0], [0.0, 1.0], [0.0, 0.0]])
logits = jnp.array([[0.2, 0.12], [0.42, 0.21], [0.22, 0.34]])
res = jnp.array(1.5022, dtype=jnp.float32)
print(cross_entropy(labels, logits)) # 0.48586553
print(optax.losses.softmax_cross_entropy(logits, labels).mean()) # 0.48586553

# test case 2
labels = jnp.array([[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0]])
logits = jnp.array([[0.8, 0.11], [0.22, 0.22], [0.1, 0.3], [0.08, 0.19]])
res = jnp.array(0.7607, dtype=jnp.float32) 
print(cross_entropy(labels, logits)) # 0.26154354 
print(optax.losses.softmax_cross_entropy(logits, labels).mean()) # 0.26154354 
```

```python
import torch

# test case 1
labels = torch.tensor([[1.0, 0.0], [0.0, 1.0], [0.0, 0.0]])
logits = torch.tensor([[0.2, 0.12], [0.42, 0.21], [0.22, 0.34]])
res = torch.tensor(1.5022, dtype=th.float32)
print(torch.nn.functional.cross_entropy(logits, labels)) # tensor(0.4859)

# test case 2
labels = torch.tensor([[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0]])
logits = tensor([[0.8, 0.11], [0.22, 0.22], [0.1, 0.3], [0.08, 0.19]])
res = torch.tensor(0.7607, dtype=th.float32)
print(torch.nn.functional.cross_entropy(logits, labels)) # tensor(0.2615)
```

**Comment by v0lta** [comment]
We used len(labels) instead of np.prod(labels.shape) in the solution to compute the numbers. Thats where the difference came from.

**Comment by v0lta** [comment]
I have removed the hardcoded numbers for this test and replaced them with a test against PyTorch's `th.nn.functional.binary_cross_entropy`. Thanks for bringing this up.
